src/unix/freebsd/disk.rs

```diff
@@ -106,14 +107,18 @@ pub unsafe fn get_all_list(container: &mut Vec<Disk>) {
             // If we have missing information, no need to look any further...
             continue;
         }
-        let fs_type: &[libc::c_char] =
-            if let Some(pos) = fs_info.f_fstypename.iter().position(|x| *x == 0) {
-                &fs_info.f_fstypename[..pos]
-            } else {
-                &fs_info.f_fstypename
-            };
-        let fs_type: &[u8] = std::slice::from_raw_parts(fs_type.as_ptr() as _, fs_type.len());
-        match fs_type {
+        let fs_type: Vec<u8> = {
+            let len = fs_info
```

Comment:
We could re-use the code from `unix/disks.rs`, never realized I was doing the same but longer here too.